app/controllers/alchemy/admin/pages_controller.rb

```diff
@@ -350,6 +353,9 @@ def redirect_path_for_switch_language
       def redirect_path_after_create_page
         if @page.redirects_to_external?
           admin_pages_path
+        elsif !@page.editable_by?(current_alchemy_user)
+          flash[:notice] = Alchemy.t('Your user role does not allow you to edit this page.')
```

Comment:
Please remove the flash message here, as we already have one for "successfully created page" and this will be lost then.